### PR TITLE
use onnx turn detector model

### PR DESCRIPTION
**Pull request by jeradf**

### CODE REVIEW

**Comment by jeradf** [comment]
this uses huggingface's optimum to load the quantized onnx model instead of the full pytorch version. inference takes ~50ms on my macbook.

File: livekit-plugins/livekit-plugins-turn-detector/livekit/plugins/turn_detector/__init__.py [hunk 1]
```diff
@@ -27,11 +27,12 @@ def __init__(self):
         super().__init__(__name__, __version__, __package__, logger)
 
     def download_files(self) -> None:
-        from transformers import AutoModelForCausalLM, AutoTokenizer
+        from optimum.onnxruntime import ORTModelForCausalLM
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-turn-detector/livekit/plugins/turn_detector/__init__.py:30]: Any way to remove the optimum dependency so we can get rid of torch? 
https://pypistats.org/packages/optimum

Comment (jeradf) [reply livekit-plugins/livekit-plugins-turn-detector/livekit/plugins/turn_detector/__init__.py:30]: i'm having some trouble getting the plain onnxruntime working so i figured we could do this in the meantime. i'll make a follow up PR that removes pytorch next week.

**Comment by davidzhao** [comment]
@jeradf I got the following when running `download-files`:

```
  File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/livekit/plugins/turn_detector/__init__.py", line 35, in download_files
    ORTModelForCausalLM.from_pretrained(HG_MODEL)
  File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/onnxruntime/modeling_ort.py", line 737, in from_pretrained
    return super().from_pretrained(
  File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/modeling_base.py", line 438, in from_pretrained
    return from_pretrained_method(
  File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/onnxruntime/modeling_decoder.py", line 609, in _from_pretrained
    return init_cls(
  File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/onnxruntime/modeling_decoder.py", line 185, in __init__
    raise ValueError(
ValueError: `use_cache` was set to `True` but the loaded model only supports `use_cache=False`. Please load your current model with `use_cache=False` or export the original model once again with `use_cache=True` when calling the `from_pretrained` method. To export your model, simply set `export=True`.

```

**Comment by davidzhao** [comment]
I got past it by setting `export=True` in
```
ORTModelForCausalLM.from_pretrained(HG_MODEL, export=True)
```
not sure if that's the right approach

**Comment by davidzhao** [comment]
the perf improvements are awesome! it feels blazing fast on my 4 year old M1 MBP.

**Comment by jeradf** [comment]
> @jeradf I got the following when running `download-files`:
> 
> ```
>   File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/livekit/plugins/turn_detector/__init__.py", line 35, in download_files
>     ORTModelForCausalLM.from_pretrained(HG_MODEL)
>   File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/onnxruntime/modeling_ort.py", line 737, in from_pretrained
>     return super().from_pretrained(
>   File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/modeling_base.py", line 438, in from_pretrained
>     return from_pretrained_method(
>   File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/onnxruntime/modeling_decoder.py", line 609, in _from_pretrained
>     return init_cls(
>   File "/Users/davidzhao/dev/livekit/agents/venv/lib/python3.10/site-packages/optimum/onnxruntime/modeling_decoder.py", line 185, in __init__
>     raise ValueError(
> ValueError: `use_cache` was set to `True` but the loaded model only supports `use_cache=False`. Please load your current model with `use_cache=False` or export the original model once again with `use_cache=True` when calling the `from_pretrained` method. To export your model, simply set `export=True`.
> ```

oops should be fixed now

**Comment by davidzhao** [comment]
woohoo, works for me now. inference is even faster.. ~50ms!

Got this warning while the model loaded.. not sure if that's expected:

```
ORTModelForCausalLM loaded a legacy ONNX model with no position_ids input, although this input is required for batched generation for the architecture llama. We strongly encourage to re-export the model with optimum>=1.14 for position_ids and batched inference support.
```

**Review by davidzhao — APPROVED** [review]
lgtm! could you add a changeset? `pnpm changeset`

**Review by theomonnom — APPROVED** [review]

**Pull request merged**